### DISCUSSION TITLE
Any convenient SQL for a quick duplication with one or more specific columns changed?

### DISCUSSION THREAD

**Discussion by KobeBryant-24** [ClickHouse/ClickHouse#69550] [Q&A] [answered]
Say, there is a table with a huge amount of columns:
(pseudo SQL below)
`create table test ( col_1 String, col_2 String, ..... , col_n DateTime, ... col_last String) engine=MergeTree() order by col_x;`

afaik, I can select columns except for some unwanted ones by `select columns('.*') except(col_n) from test`.
also, I can simply duplicate data by `insert into test select * from test`.
however, I cannot find a way to make the `col_n` randomized when the data is duplicated.
I tried `insert into test select col_n, columns('.*') except(col_n) from test` but obviously failed since the column order is incorrect.
I wonder if there is a function to achieve like `select columns(before col_n), col_n, columns(after col_n)`, thanks in advance

**Comment by SaltTan** [comment, marked as answer]
You can use `replace`
`select * replace(expression as col_n) from test`

**Reply by KobeBryant-24** [reply]
Appreciated!!
